src/main/java/nl/hsac/fitnesse/fixture/Environment.java

```diff
@@ -250,6 +250,20 @@ public void doHttpPost(String url, String templateName, Object model, HttpRespon
     public void doHttpPost(String url, HttpResponse result, Map<String, Object> headers, String contentType) {
         httpClient.post(url, result, headers, contentType);
     }
+
+    /**
+     * Performs POST to supplied url of result's request.
+     * @param url url to post to.
+     * @param result result containing request, its response will be filled.
+     * @param headers headers to add.
+     * @param contentType contentType for request.
+     */
+    public void doHttpFilePost(String url, HttpResponse result, Map<String, Object> headers, String contentType, String fileName) {
+        String filesDir = getFitNesseFilesSectionDir();
```

Comment:
Can you move the logic to deal with full paths, or urls pointing to the files section to the fixture. So that the environment just deals with a java File reference